hivemind_bus_client/http_client.py

```
@@ -0,0 +1,388 @@
+                 password: Optional[str] = None,
+                 crypto_key: Optional[str] = None,
+                 host: Optional[str] = None,
+                 port: Optional[int] = None,
+                 useragent: str = "HiveMindHTTPClientV1.0",
+                 self_signed: bool = True,
+                 share_bus: bool = False,
+                 compress: bool = True,
+                 binarize: bool = True,
+                 identity: NodeIdentity = None,
+                 internal_bus: Optional[OVOSBusClient] = None,
+                 bin_callbacks: BinaryDataCallbacks = BinaryDataCallbacks()):
```

Ruff (0.8.2)

Do not perform function call `BinaryDataCallbacks` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)